### PR TITLE
cli-test-utils: print_stream helper function

### PR DESCRIPTION
**Pull request by liamaharon**
- Adds a new `print_stream` function, closing https://github.com/paritytech/substrate/issues/13985
- Updates fn `common::start_node` to return a `tokio::process::Child` so it can be used with `print_stream`
- Updates fn `common::start_node` to return the `ws_url` of the node it spawned, so its stderr doesn't need to be consumed to derive it later
- Updates `wait_n_finalized_blocks` to retry the WS connection a few times before giving up. The new async Child is not always available to connect to WS immediately after spawns

### CODE REVIEW

**Comment by brunopgalvao** [comment]
When I run the following on this branch:
```
cargo test --features try-runtime -p try-runtime-cli -- --nocapture
```

I get:
```
   Compiling try-runtime-cli v0.10.0-dev (/Users/bruno/src/temp/delete/substrate/utils/frame/try-runtime/cli)
    Finished test [unoptimized + debuginfo] target(s) in 3m 25s
     Running unittests src/lib.rs (target/debug/deps/try_runtime_cli-4c611e9fb746222d)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/follow_chain.rs (target/debug/deps/follow_chain-46b53fb1de696150)

running 1 test
test tests::follow_chain_works has been running for over 60 seconds
thread 'tests::follow_chain_works' panicked at 'called `Result::unwrap()` on an `Err` value: "Could not create rpc client after 30 seconds"', utils/frame/try-runtime/cli/tests/follow_chain.rs:48:63
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test tests::follow_chain_works ... FAILED

failures:

failures:
    tests::follow_chain_works

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 360.20s

error: test failed, to rerun pass `-p try-runtime-cli --test follow_chain`
```
I've tried increasing the time limit here: 
https://github.com/paritytech/substrate/blob/438e887909f81b270b050b886d703810139e78fc/utils/frame/try-runtime/cli/tests/follow_chain.rs#L34
To `15 * 60` (I have an M1) and that resulted in the same error.

**Comment by liamaharon** [comment]
@brunopgalvao which branch are you using?

**Comment by brunopgalvao** [comment]
> @brunopgalvao which branch are you using?

liam-cli-test-utils-print-stream

File: utils/frame/try-runtime/cli/tests/follow_chain.rs [hunk 1]
```diff
@@ -47,9 +44,8 @@ mod tests {
 			}
 
 			// Start a node and wait for it to begin finalizing blocks
-			let mut node = common::KillChildOnDrop(common::start_node());
-			let ws_url = common::extract_info_from_output(node.stderr.take().unwrap()).0.ws_url;
-			common::wait_n_finalized_blocks(1, &ws_url).await;
+			let (_child, ws_url) = common::start_node();
```
Comment (liamaharon) [inline utils/frame/try-runtime/cli/tests/follow_chain.rs:47]: For some reason, if I don't specify _child like this: 

```suggestion
			let (_, ws_url) = common::start_node();
```

Something happens to the node and it doesn't accept WS requests 🤷 

@brunopgalvao this should fix the issue you were having

**Review by brunopgalvao — APPROVED** [review]

**Comment by liamaharon** [comment]
Seems this is not necessary, if you don't specify `.stdout(...)` when creating the child, it seems to auto direct output to your terminal.

cc @brunopgalvao

**Pull request closed without merging**